src/_pytest/python.py

```diff
@@ -1096,6 +1097,36 @@ def _validate_if_using_arg_names(self, argnames, indirect):
                         pytrace=False,
                     )
 
+    def _validate_explicit_parameters(self, argnames, indirect):
+        """
+        The argnames in *parametrize* should either be declared explicitly via
+        indirect list or explicitly in the function
+
+        :param List[str] argnames: list of argument names passed to ``parametrize()``.
+        :param indirect: same ``indirect`` parameter of ``parametrize()``.
+        :raise ValueError: if validation fails
+        """
+        func_name = self.function.__name__
+        if type(indirect) is bool and indirect is True:
+            return
+        parametrized_argnames = list()
+        funcargnames = _pytest.compat.getfuncargnames(self.function)
+        if type(indirect) is list:
```

Comment:
Don't forget to check using `isinstance(indirect, Iterable)` here 😉

Reply:
Isn't the check in 1114 sufficient? I thought that only arguments of type `list` or `bool` might be passed to this function

Reply:
Yes, but that doesn't prevent the user from passing something else...

Actually I think we better check if `indirect` is valid (either a `bool` or an iterable) just before line 975... what do you think?

Reply:
I think the check we're interested in is done in `_resolve_arg_value_types`, line 1093 This function is called just below I wrote the call for `_validate_explicit_parameters` (line 1003). I have a feeling it might be suifficient to move `_validate_explicit_parameters` two lines below to avoid unnecessary checks. Does it sound right?